pkg/cmd/server/origin/auth_config.go

```diff
@@ -101,18 +105,20 @@ func BuildAuthConfig(options configapi.MasterConfig) (*AuthConfig, error) {
 		UserRegistry:     userRegistry,
 
 		SessionAuth: sessionAuth,
+
+		HandlerWrapper: sessionHandlerWrapper,
 	}
 
 	return ret, nil
 }
 
-func BuildSessionAuth(secure bool, config *configapi.SessionConfig) (*session.Authenticator, error) {
+func BuildSessionAuth(secure bool, config *configapi.SessionConfig) (*session.Authenticator, handlerWrapper, error) {
```

Comment:
Returning private type feels slightly wierd on public function.  Fine here though for now.